### PR TITLE
[Chore] Add issue link tempate for IDEA

### PR DESCRIPTION
**Pull request by Gallardot**
## Purpose of the pull request

With IDEA's issue tracker, we can effortlessly navigate to Github issues or pull requests, facilitating a better understanding of the change history for developers.

This has numerous benefits for IDEA, such as in comments and in version control views.

Here's the link to the documentation for IDEA Git integration with the issue tracker: https://www.jetbrains.com/help/idea/handling-issues.html

Through this pull request, issues can be accurately navigated to Github.

Before:

image

After:

image

## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

(or)

If your pull request contain incompatible change, you should also add it to `docs/docs/en/guide/upgrede/incompatible.md`

### CODE REVIEW

**Review by SbloodyS — COMMENTED** [review]
Considering that this is more about user convenience and not much help to the project. So I'm +0 on this. See if anyone else has another opinion.

File: .idea/vcs.xml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+          <option name="issueRegexp" value="#(\d+)" />
+          <option name="linkRegexp" value="https://github.com/apache/dolphinscheduler/issues/$1" />
+        </IssueNavigationLink>
+      </list>
+    </option>
+  </component>
+  <component name="VcsDirectoryMappings">
+    <mapping directory="" vcs="Git" />
+  </component>
+</project>
```
Comment (ruanwenjun) [inline .idea/vcs.xml:16]: Seems this will case conflicts If the dev users already have a `vcs.xml` file at local?

Comment (Gallardot) [reply .idea/vcs.xml:16]: The default configuration for this file is as follows:
```
<?xml version="1.0" encoding="UTF-8"?>
<project version="4">
  <component name="VcsDirectoryMappings">
    <mapping directory="" vcs="Git" />
  </component>
</project>
```
Most developers' configuration files should look like this.

I have now only added one component:
```
  <component name="IssueNavigationConfiguration">
    <option name="links">
      <list>
        <IssueNavigationLink>
          <option name="issueRegexp" value="#(\d+)" />
          <option name="linkRegexp" value="https://github.com/apache/dolphinscheduler/issues/$1" />
        </IssueNavigationLink>
      </list>
    </option>
  </component>
```
I am not sure if other developers will modify this file to implement other functionalities.

With IDEA's issue tracker, we can effortlessly navigate to Github issues or pull requests, facilitating a better understanding of the change history for developers.

Comment (ruanwenjun) [reply .idea/vcs.xml:16]: Ok.

**Review by ruanwenjun — APPROVED** [review]
+1 to this pr.

It makes using the `IDEA` fun!

**Review by davidzollo — APPROVED** [review]
+1
good job

**Pull request merged**